### ISSUE TITLE
App panic when clicking on a notification panel if there's a systray icon

### ISSUE THREAD

**Issue by theHilikus** [fyne-io/fyne#4385]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When I click on a gnome notification panel when there's a systray icon, the app crashes with 

```
❯ go run .
2023/11/13 18:54:00 Fyne error:  Preferences API requires a unique ID, use app.NewWithID() or the FyneApp.toml ID field
2023/11/13 18:54:00   At: /home/hilikus/go/pkg/mod/fyne.io/fyne/v2@v2.4.1/app/app.go:60
2023/11/13 18:54:00 Failed to read icon format image: unknown format
panic: runtime error: index out of range [2] with length 2

goroutine 82 [running]:
fyne.io/systray.stayRegistered()
        /home/hilikus/dev/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230722100817-88df1e0ffa9a/systray_unix.go:273 +0x31a
created by fyne.io/systray.nativeStart in goroutine 1
        /home/hilikus/dev/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230722100817-88df1e0ffa9a/systray_unix.go:230 +0xbbc
exit status 2
```

If I remove the systray code and click on the notification panel there is no panic

### How to reproduce

1. Create a systray icon
2. Create a notification
3. Click on the notification

### Example code

```go
	app := app.New()

	window := app.NewWindow("Crasher")
	window.Resize(fyne.NewSize(400, 600))

	if desk, ok := app.(desktop.App); ok {
		showItem := fyne.NewMenuItem("Show", func() {
			window.Show()
		})
		menu := fyne.NewMenu("Menu", showItem)
		desk.SetSystemTrayMenu(menu)
		systray.SetTitle("Crash")
	}

	app.SendNotification(fyne.NewNotification("click me", "crash me"))

	window.SetContent(widget.NewLabel("Hello World!"))
	window.ShowAndRun()
```

### Fyne version

2.4.1 & a289a25

### Go compiler version

1.21.4

### Operating system and version

Ubuntu 22.04.3 LTS

**Comment by andydotxyz** [comment]
I upgraded the systray dep, this should be fixed now.

**Comment by theHilikus** [comment]
Can confirm the crash is fixed on 4128839

Thank you Andy
